app/com/arpnetworking/metrics/portal/reports/Sender.java

```diff
@@ -31,12 +33,16 @@
     /**
      * Send some {@link RenderedReport} documents to a {@link Recipient}.
      *
+     * @param report The report being sent.
      * @param recipient The recipient to notify.
-     * @param formatsToSend The reports to send.
+     * @param formatsToSend The reports to send. Must be non-empty.
+     * @param scheduled The instant that the report being sent was generated for.
      * @return A CompletionStage that completes when the transmission has completed.
      */
     CompletionStage<Void> send(
+            Report report,
             Recipient recipient,
-            ImmutableMap<ReportFormat, RenderedReport> formatsToSend
+            ImmutableMap<ReportFormat, RenderedReport> formatsToSend,
+            Instant scheduled
```

Comment:
I'm a little bit uncomfortable about how I keep needing to add arguments to interface methods: it seems bad for pluggability, since people who drop in their own `Sender` implementations will need to rewrite them whenever the interface changes.

The obvious solution is to replace all of these arguments with a single omnibus object whose class is named, like, `ArgumentsToSend`. But that seems pretty gross in its own way, for reasons I can't really put my finger on.

Thoughts on this, @vjkoskela or @BrandonArp ?

Reply:
As discussed.